### PR TITLE
fix(kuma-cp): change way of setting if resource is read only

### PR DESCRIPTION
**Pull request by lukidzi**
Changed how we check if a resource should be read-only. Previously we were allowing only Dataplane resources to be edited on the zone, now we are checking if the resource is synchronized and in which direction.

If we are on Global:
* and resource is synchronized from Global
* or resource is not synchronized
* or resource is synchronized from Global and from Zone
we allow you to edit it from global.

If we are on Zone:
* and resource is synchronized from Zone
* or resource is not synchronized
* or resource is synchronized from Global and from Zone
we allow you to edit it from Zone.

- [X] Link to docs PR or issue --
- [X] Link to UI issue or PR --
- [X] Is the [issue worked on linked][1]? --
- [X] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [X] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Unit Tests --
- [X] E2E Tests --
- [X] Manual Universal Tests --
- [X] Manual Kubernetes Tests --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/api-server/server.go [hunk 1]
```diff
@@ -264,6 +264,12 @@ func addResourcesEndpoints(ws *restful.WebService, defs []model.ResourceTypeDesc
 	}
 }
 
+func ShouldBeReadOnly(kdsFlag model.KDSFlagType, cfg *kuma_cp.Config) bool {
+	return cfg.ApiServer.ReadOnly ||
+		(cfg.Mode == config_core.Global && kdsFlag.Has(model.FromZoneToGlobal) && !kdsFlag.Has(model.FromGlobalToZone)) ||
```
Comment (lahabana) [inline pkg/api-server/server.go:269]: `kdsFlag.Has(model.FromZoneToGlobal)` and `kdsFlag.Has(model.FromGlobalToZone)` are not going to be set for resources that are not synced at all (like RBAC). I think something more correct is:

```go
if cfg.ApiServer.ReadOnly {
   return true
}
if kdsFlag == model.KDSDisabled { // Add KDSDisabled = 0 to model
   return false 
}
if cfg.Mode == config_core.Global && !kdsFlag.Has(model.ProvidedByGlobal) {
  return true
}
if cfg.Mode == config_core.Zone && !kdsFlag.Has(model.ProvidedByZone) {
  return true
}
return false

**Review by lahabana — CHANGES_REQUESTED** [review]

File: pkg/api-server/server.go (hunk 1 above)
Comment (lukidzi) [reply pkg/api-server/server.go:269]: It looks better I found only one thing that looks strange
https://github.com/kumahq/kuma/blob/master/pkg/core/resources/apis/mesh/zz_generated.resources.go#L2806

Comment (lahabana) [reply pkg/api-server/server.go]: Yes these go both ways (From global to zone and from zone to global)

**Comment by lukidzi** [comment]
@Mergifyio backport release-2.0

**Pull request merged**